hive-metastore/src/main/java/org/apache/iceberg/hive/MetastoreUtil.java

```diff
@@ -37,6 +53,17 @@ public static boolean hive3PresentOnClasspath() {
     return HIVE3_PRESENT_ON_CLASSPATH;
   }
 
+  /**
+   * Invokes alterTable via metaStoreClient without triggering recursive listing.
+   * ref: https://github.com/apache/iceberg/pull/734#issuecomment-574062407
```

Comment:
Rather than referencing the issue, it would be more clear to just state the resolution: if possible, call the alter_table method with an environment context that turns off stats updates.